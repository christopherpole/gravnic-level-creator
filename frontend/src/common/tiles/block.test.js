import React from 'react';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import BlockTile from './block';

configure({ adapter: new Adapter() });

describe('<BlockTile />', () => {
  let props;

  beforeEach(() => {
    props = {
      color: '#ff0000',
    };
  });

  it('Renders without exploding', () => {
    const blockTile = shallow(<BlockTile {...props} />);

    expect(blockTile).toHaveLength(1);
  });

  it('Matches the current snapshot', () => {
    const blockTile = shallow(<BlockTile {...props} />);
    expect(toJson(blockTile)).toMatchSnapshot();
  });
});
