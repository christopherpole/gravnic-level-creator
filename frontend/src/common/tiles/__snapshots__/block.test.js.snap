// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlockTile /> Matches the current snapshot 1`] = `
<styled.div
  color="#ff0000"
  moveSpeed={0}
  shrinking={false}
/>
`;
